File: arena-backend/src/entities/group-channel.entity.ts
import {
  Column,
  Entity,
  JoinColumn,
  ManyToOne,
  OneToOne,
  PrimaryColumn,
} from "typeorm";
import { ChannelEntity } from "./channel.entity";
import { FileEntity } from "./file.entity";

@Entity({ name: "group_channels" })
export class GroupChannelEntity {
  @PrimaryColumn({ type: "text" })
  channelId: string;

  @OneToOne(() => ChannelEntity)
  @JoinColumn({ name: "channelId", referencedColumnName: "channelId" })
  channel: ChannelEntity;

  @Column({ type: "text", nullable: true })
  iconFileId: string | null;

  @ManyToOne(() => FileEntity, { nullable: true })
  @JoinColumn({ name: "iconFileId", referencedColumnName: "fileId" })
  icon: FileEntity | null;
}

File: legacy/backend/src/main/java/app/sandori/arena/api/domain/channel/GroupChannelEntity.java
package app.sandori.arena.api.domain.channel;

import app.sandori.arena.api.domain.file.FileEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "group_channels")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class GroupChannelEntity {

    @Id
    private String channelId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "channelId", referencedColumnName = "channelId", insertable = false, updatable = false)
    private ChannelEntity channel;

    @Column
    private String iconFileId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "iconFileId", referencedColumnName = "fileId", insertable = false, updatable = false)
    private FileEntity icon;

    public GroupChannelEntity(String channelId, String iconFileId) {
        this.channelId = channelId;
        this.iconFileId = iconFileId;
    }
}
